services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -235,21 +236,22 @@ open class HankeServiceImpl(
         hankeLoggingService.logDelete(hanke, userId)
     }
 
-    private fun initAccessForCreatedHanke(hanke: Hanke, userId: String) {
-        val hankeId = hanke.id!!
-        val permissionAll = permissionService.setPermission(hankeId, userId, Role.KAIKKI_OIKEUDET)
-        val perustaja =
-            hanke.perustaja ?: throw HankeArgumentException("Missing perustaja information")
-        hankeKayttajaService.addHankeFounder(hankeId, perustaja, permissionAll)
-        hankeKayttajaService.saveNewTokensFromHanke(hanke)
-    }
-
     private fun anyHakemusProcessingInAllu(hakemukset: List<Application>): Boolean =
         hakemukset.any {
             logger.info { "Hakemus ${it.id} has alluStatus ${it.alluStatus}" }
             !applicationService.isStillPending(it)
         }
 
+    private fun initAccessForCreatedHanke(hanke: Hanke, userId: String) {
+        val hankeId = hanke.id!!
+        val permissionAll = permissionService.setPermission(hankeId, userId, Role.KAIKKI_OIKEUDET)
+        val perustaja = hanke.perustaja
+        if (perustaja != null) {
+            hankeKayttajaService.addHankeFounder(hankeId, perustaja, permissionAll)
```

Comment:
So this still creates a user for every created hanke if the caller gives a perustaja?

Reply:
Idea is that the perustaja is defined by the backend (before this step). It is not yet implemented as this pr will only focus on johtoselvitys/hanke generation. At least that is how I understood things based on the discussion last week. 

I can add a validator to make sure that the client does not set this field.

Reply:
Ah yes, I didn't notice that the controller wiped the perustaja. I don't know if this is something the controller should be doing, it feels very much like business logic.

We could really use a separate DTO for hanke creation since there are so many fields we need to ignore.

Do we need perustaja in Hanke? With the current understanding, we don't want to add it to any responses and we don't want it to be modifiable with PUT requests. We could add it directly to the entity in `createHanke` and then forget about it. This would require we add `Perustaja?` as a separate parameter to `createHanke`. I think this would also make the logic surrounding the perustaja more transparent. For now, we could make the perustaja null in the controller call. When adding the hankekayttaja for all hanke, we could add a method in front of createHanke that finds the perustaja from the contacts.

Reply:
A separate dto for these requests would be a major improvement, I agree.

I removed Perustaja from the domain object as suggested.